src/pages/iou/request/IOURequestStartPage.js

```diff
@@ -0,0 +1,161 @@
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import DragAndDropProvider from '@components/DragAndDrop/Provider';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import TabSelector from '@components/TabSelector/TabSelector';
+import transactionPropTypes from '@components/transactionPropTypes';
+import useLocalize from '@hooks/useLocalize';
+import usePrevious from '@hooks/usePrevious';
+import * as DeviceCapabilities from '@libs/DeviceCapabilities';
+import * as IOUUtils from '@libs/IOUUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import OnyxTabNavigator, {TopTab} from '@libs/Navigation/OnyxTabNavigator';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as TransactionUtils from '@libs/TransactionUtils';
+import reportPropTypes from '@pages/reportPropTypes';
+import styles from '@styles/styles';
+import * as IOU from '@userActions/IOU';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import IOURequestStepAmount from './step/IOURequestStepAmount';
+import IOURequestStepDistance from './step/IOURequestStepDistance';
+import IOURequestStepRoutePropTypes from './step/IOURequestStepRoutePropTypes';
+import IOURequestStepScan from './step/IOURequestStepScan';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: IOURequestStepRoutePropTypes.isRequired,
+
+    /* Onyx Props */
+    /** The report that holds the transaction */
+    report: reportPropTypes,
+
+    /** The tab to select by default (whatever the user visited last) */
+    selectedTab: PropTypes.oneOf(_.values(CONST.TAB_REQUEST)),
+
+    /** The transaction being modified */
+    transaction: transactionPropTypes,
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+};
+
+const defaultProps = {
+    report: {},
+    selectedTab: CONST.TAB_REQUEST.SCAN,
+    transaction: {},
+    betas: [],
+};
+
+function IOURequestStartPage({
+    report,
+    route,
+    route: {
+        params: {iouType, reportID},
+    },
+    selectedTab,
+    transaction,
+    betas,
+}) {
+    const {translate} = useLocalize();
+    const [isDraggingOver, setIsDraggingOver] = useState(false);
+    const tabTitles = {
+        [CONST.IOU.TYPE.REQUEST]: translate('iou.requestMoney'),
+        [CONST.IOU.TYPE.SEND]: translate('iou.sendMoney'),
+        [CONST.IOU.TYPE.SPLIT]: translate('iou.splitBill'),
+    };
+    const transactionRequestType = useRef(TransactionUtils.getRequestType(transaction));
+    const previousIOURequestType = usePrevious(transactionRequestType.current);
+
+    // Clear out the temporary money request when this component is unmounted
+    useEffect(
+        () => () => {
+            IOU.startMoneyRequest_temporaryForRefactor('');
```

Comment:
Hm this just sets the reportID of the draft transaction to an empty string. Do we need to actually clear it?

Reply:
Good catch. I think this was a mistake. Its intention is to clear the request entirely. I've updated the code to add a `clearMoneyRequest(transactionID)` and use that instead.